pkg/operator/ceph/cluster/watcher.go

```diff
@@ -94,20 +108,16 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, nodeName)
```

Comment:
What about the case when there is no error and `osds` string returns empty.  Shouldn't we return `true` in that case as well ?

Reply:
Yes, we should check for empty string case.